wyvern/wyvern_request.py

```diff
@@ -163,3 +179,24 @@ def get_original_identifier(
                 identifier=primary_identifier_key,
                 feature_name=feature_name,
             )
+
+    def add_shadow_request_call(
+        self,
+        shadow_request: ShadowRequest,
+    ):
+        if self.shadow_requests is None:
+            self.shadow_requests = []
+
+        self.shadow_requests.append(shadow_request)
+
+    async def execute_shadow_requests(self):
+        if self.shadow_requests is None:
+            return
+        try:
+            request_context.set(self)
```

Comment:
hmmmmmmm. I'm not so sure this is going to cause raise condition with the next fastapi request if the current request session is "done".

let's watch out for it